### ISSUE TITLE
What is the correct syntax for compile overrides that are qualified?

### ISSUE THREAD

**Issue by zkarj735** [IBM/ibmi-bob#211]
I cannot work out the correct syntax to override compile parameters that use qualified object names, such as BNDDIR, HLPPNLGRP, and PGM.

It seems to understand enough to strip away the library, as when I write
```
C_STDAUT.CMD: private PGM := AJENKS/C_STDAUT
```
The output shows the command with `PGM(C_STDAUT)`.

What I actually want to do is
```
C_STDAUT.CMD: private PGM := &BUILDLIB/C_STDAUT
```

**Comment by zkarj735** [comment]
Actually, my problem is more fundamental. I cannot get my overrides to work at all now.
```
C_STDAUT.CMD: private PGM := JIM
C_STDAUT.CMD: private HLPPNLGRP := BOB
C_STDAUT.CMD: c_stdaut.cmdsrc
C_STDAUT.PGM: c_stdaut.pgm.clle
C_STDAUT.PNLGRP: c_stdaut.pnlgrpsrc
```
Results in
```
Current library: AJENKS
Library list: ICMSSW QGPL QTEMP
Commands:
		err=*EVENTF /QOpenSys/pkgs/bin/makei compile -f c_stdaut.cmdsrc -e BUILDLIB=AJENKS

Set variable <BUILDLIB> to 'AJENKS'
targets: C_STDAUT.CMD
> /QOpenSys/pkgs/bin/make -k BUILDVARSMKPATH="/tmp/tmp4waqjmux" -k BOB="/QOpenSys/pkgs/lib/bob" -f "/QOpenSys/pkgs/lib/bob/src/mk/Makefile" C_STDAUT.CMD
=== Creating command [c_stdaut.cmdsrc]
CRTCMD CMD(AJENKS/C_STDAUT) srcstmf('/builds/ibmi-utility/compile-tools/c_stdaut.cmdsrc') PGM(C_STDAUT) VLDCKR(*NONE) PMTFILE(*NONE) HLPPNLGRP(C_STDAUT) HLPID(C_STDAUT) AUT() ALLOW() TEXT('')
✓ C_STDAUT.CMD was created successfully!

Objects:             0 failed 1 succeed 1 total
Build Completed!

Fetching errors from AJENKS/c_stdaut.
```

**Comment by zkarj735** [comment]
I noticed issue #194 and tried some further variations.
```
C_STDAUT.CMD: PGM = AJENKS/C_STDAUT
C_STDAUT.CMD: HLPPNLGRP = AJENKS/C_STDAUT
C_STDAUT.CMD: c_stdaut.cmdsrc
```
Showed the following output
```
Current library: AJENKS
Library list: ICMSSW QGPL QTEMP
Commands:
		err=*EVENTF /QOpenSys/pkgs/bin/makei compile -f c_stdaut.cmdsrc -e BUILDLIB=AJENKS

Set variable <BUILDLIB> to 'AJENKS'
targets: C_STDAUT.CMD
> /QOpenSys/pkgs/bin/make -k BUILDVARSMKPATH="/tmp/tmpsa20txci" -k BOB="/QOpenSys/pkgs/lib/bob" -f "/QOpenSys/pkgs/lib/bob/src/mk/Makefile" C_STDAUT.CMD
=== Creating command [c_stdaut.cmdsrc]
CRTCMD CMD(AJENKS/C_STDAUT) srcstmf('/builds/ibmi-utility/compile-tools/c_stdaut.cmdsrc') PGM(C_STDAUT) VLDCKR(*NONE) PMTFILE(*NONE) HLPPNLGRP(AJENKS/C_STDAUT) HLPID(C_STDAUT) AUT() ALLOW() TEXT('')
✓ C_STDAUT.CMD was created successfully!

Objects:             0 failed 1 succeed 1 total
Build Completed!

Fetching errors from AJENKS/c_stdaut.
```
It looks like `HLPPNLGRP` has worked but `PGM` has not. Yet, on inspecting the resulting command I note:
```
Program to process command . . . . . . :   C_STDAUT
  Library  . . . . . . . . . . . . . . :     *LIBL 
Help panel group . . . . . . . . . . . :   C_STDAUT
  Library  . . . . . . . . . . . . . . :     *LIBL 
```

**Comment by edmundreinhardt** [comment]
To qualify where an object is being built.  One should use the 'objlib' setting of the iproj.json file.
Note that this can be overridden for any subdirectory using using the .ibmi.json file.
https://ibm.github.io/ibmi-bob/#/prepare-the-project/iproj-json?id=objlib
https://ibm.github.io/ibmi-bob/#/prepare-the-project/ibmi-json?id=buildobjlib

BTW BOB prefers the = assignment rather than the := assignment.  That is the fundamental issue behind issue #194

**Comment by edmundreinhardt** [comment]
Let me know if this clears things up?

**Comment by edmundreinhardt** [comment]
FYI the := limitation has been fixed in Bob 2.4.11
But even so, I think you should be using `objlib` in iproj.json to specify the target object library.

**Comment by zkarj735** [comment]
I would expect to be able to specify any qualified or list value in any command parameter I can override. It is entirely conceivable that a compile/create command would need to reference an object in a library that is not the build library.

The double issue that exists currently is that this does not seem to be supported, but the build output implies some parameters _do_ support it when they don't.
